File: android/src/main/java/com/getcapacitor/community/classes/events/PeerIDChangedEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;

public class PeerIDChangedEvent extends PeerIDEvent {

    @Nullable
    private final String oldPeerID;

    @NonNull
    private final String message;

    public PeerIDChangedEvent(@NonNull String peerID, @Nullable String oldPeerID, @NonNull String message) {
        super(peerID);
        this.oldPeerID = oldPeerID;
        this.message = message;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = super.toJSObject();

        if (oldPeerID != null) {
            result.put("oldPeerID", oldPeerID);
        }

        result.put("message", message);

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Events/PeerIDChangedEvent.swift
import Foundation
import Capacitor

@objc public class PeerIDChangedEvent: PeerIDEvent {
    let oldPeerID: String?
    let message: String

    init(_ peerID: String, _ oldPeerID: String?, _ message: String) {
        self.oldPeerID = oldPeerID
        self.message = message

        super.init(peerID)
    }

    override public func toJSObject() -> JSObject {
        var result = super.toJSObject()

        if let oldPeerID = oldPeerID {
            result["oldPeerID"] = oldPeerID
        }

        result["message"] = message

        return result
    }
}
